internal/appconfig/validation.go

```diff
@@ -1,14 +1,53 @@
 package appconfig
 
 import (
+	"context"
+	"errors"
 	"fmt"
 	"reflect"
 	"strings"
 
 	"github.com/go-playground/validator/v10"
+	"github.com/superfly/flyctl/client"
+	"github.com/superfly/flyctl/terminal"
 )
 
-func (c *Config) Validate() (err error) {
+func (cfg *Config) Validate(ctx context.Context) (err error) {
+	appNameFromContext := NameFromContext(ctx)
+
+	apiClient := client.FromContext(ctx).API()
+	appCompact, err := apiClient.GetAppCompact(ctx, appNameFromContext)
+
+	if err != nil {
+		return err
+	}
+
+	if appCompact.PlatformVersion == MachinesPlatform {
+		return cfg.validateLocally()
+
+	} else {
+		parsedCfg, err := apiClient.ParseConfig(ctx, appNameFromContext, cfg.SanitizedDefinition())
+		if err != nil {
+			return err
+		}
+
+		if !parsedCfg.Valid {
+			fmt.Println()
+			if len(parsedCfg.Errors) > 0 {
+				terminal.Errorf("\nConfiguration errors in %s:\n\n", cfg.ConfigFilePath())
```

Comment:
wdyt if if we avoid terminal output in this method, and instead return an error suggesting `fly config validate` to see details.

Reply:
I totally agree, i just moved it into that function since that's how we did things in the places I got the function